packages/core/src/utils/LockManager.ts

```diff
@@ -0,0 +1,47 @@
+import { v4 as uuidv4 } from 'uuid';
+
+class LockManager {
+  private static _instance: LockManager;
+  private locks: Map<string, boolean>;
+
+  private constructor() {
+    this.locks = new Map();
+  }
+
+  public static get instance() {
+    if (!LockManager._instance) {
+      LockManager._instance = new LockManager();
+    }
+
+    return LockManager._instance;
+  }
+
+  createLock(): string {
+    const lockId = uuidv4();
+    this.locks.set(lockId, true);
+    return lockId;
+  }
+
+  deleteLock(lockId: string): void {
+    this.locks.delete(lockId);
+  }
+
+  deleteAllLocks(): void {
+    this.locks.clear();
+  }
+
+  waitForLockRelease(): Promise<void> {
+    return new Promise((resolve) => {
+      const checkLocks = () => {
+        if (this.locks.size === 0) {
+          resolve();
+        } else {
+          setTimeout(checkLocks, 100);
```

Comment:
How is `100` chosen? By experiment or? Any specific reason?

Reply:
Lowering the `setTimeout` increases the frequency of lock checks, which may improve responsiveness but increases computation. I have decreased the value from 100 to 10, and the CPU usage for 70 diagrams increased by around 3%, with a 0.2s decrease in compilation time.

Reply:
ok we can keep 100